File: appacademy/sandwich/sandwich.rb
class Butler 
    def get_peanut_butter
        sleep 1
        return "peanut butter"
    end

    def get_jelly
        sleep 1
        return "jelly"
    end
end

class Customer
    def get_sandwich(butler1, butler2)
        ingredient_one = butler1.get_peanut_butter
        ingredient_two = butler2.get_jelly

        puts "putting together #{ingredient_one} and #{ingredient_two}"
    end
end

w1 = Butler.new
w2 = Butler.new
c = Customer.new

t1 = Time.new
c.get_sandwich(w1, w2)
t2 = Time.new
puts "took #{t2 - t1} seconds"

File: womenwhogo/sandwich/sandwich.js
function makeSandwich() {
    const t1 = new Date;
    const pb = getPeanutButter();
    const j = getJelly();
    console.log(`Sandwich is ready! Putting together ${pb} and ${j}`);
    const t2 = new Date;
    const diff = (t2.getTime() - t1.getTime()) / 1000;
    console.log(`sandwich took ${diff} seconds to make`);
}

function getPeanutButter() {
    console.log('Butler 1 says: I am getting the peanut butter');
    simulateWork();
    return 'peanut butter';
}

function getJelly() {
    console.log('Butler 2 says: I am getting the jelly');
    simulateWork();
    return 'jelly';
}

function simulateWork() {
    const t1 = new Date;
    for (let t2 = new Date; ((t2.getTime() - t1.getTime()) / 1000) < 1; t2 = new Date) { }
}


makeSandwich();